packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -465,21 +466,32 @@ export async function upgradeAsync(
       await maybeUpgradeEmulatorAsync();
     }
   } else if (!targetSdkVersion) {
-    if (program.nonInteractive) {
-      log.warn(
-        `You provided the target SDK version value of ${targetSdkVersionString} which does not seem to exist. Upgrading anyways because the command is running in nonInteractive mode.`
-      );
-    } else {
-      // If they provide an apparently unsupported sdk version then let people try
-      // anyways, maybe we want to use this for testing alpha versions or
-      // something...
-      const answer = await confirmAsync({
-        message: `You provided the target SDK version value of ${targetSdkVersionString} which does not seem to exist. But hey, I'm just a program, what do I know. Do you want to try to upgrade to it anyways?`,
-      });
+    // This is useful when testing the beta internally, before actually
+    // releasing it as a public beta. At this point, we won't have "beta" set on
+    // the versions endpoint and so Versions.releasedSdkVersionsAsync will not
+    // return the beta version, even with the EXPO_BETA flag set.
+    if (getenv.boolish('EXPO_BETA', false)) {
+      const allSdkVersions = await Versions.sdkVersionsAsync();
+      targetSdkVersion = allSdkVersions[targetSdkVersionString];
+    }
 
-      if (!answer) {
-        return;
-      }
+    // If we still don't have a version, even after searching through unreleased versions
+    // when the EXPO_BETA flag is set, then we will just bail out because this is not
+    // going to work!
+    if (!targetSdkVersion) {
+      const sdkVersionNumbers = Object.keys(sdkVersions).map(value => parseInt(value, 10));
+      const minSdkVersion = sdkVersionNumbers.reduce(
+        (val, acc) => Math.min(val, acc),
+        sdkVersionNumbers[0]
+      );
+      const maxSdkVersion = sdkVersionNumbers.reduce(
+        (val, acc) => Math.max(val, acc),
+        sdkVersionNumbers[0]
+      );
+      throw new CommandError(
+        `You provided the target SDK version value of ${targetSdkVersionString}, which does not seem to exist.\n` +
+          `Valid SDK versions are in the range of ${minSdkVersion}.0.0 to ${maxSdkVersion}.0.0.`
```

Comment:
printing a range here because providing a list of SDK versions makes the error hard to read and will be even harder to read over time. might want to generalize to other places where we do the same thing